CRM/CiviAwards/BAO/AwardDetail.php

```diff
@@ -76,4 +77,42 @@ private static function validateDates(array $params) {
     }
   }
 
+
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array $params
+   *  Request Parameters
+   *
+   * @throws \Exception
+   */
+  private static function validateReviewFields(array &$params) {
+    $reviewFieldsLegalOptions = [
+      'id' => FILTER_VALIDATE_INT,
+      'weight' => FILTER_VALIDATE_INT,
+      'required' => FILTER_VALIDATE_BOOLEAN
+    ];
+    $reviewFields = CRM_Utils_Array::value('review_fields', $params);
+    if (is_array($reviewFields)) {
+      foreach ($reviewFields as &$reviewFieldOptions) {
```

Comment:
Do we need to pass by reference here?

Reply:
I want to modify the array itself with filtered values instead making a new copy of the array again